core/embed/trezorhal/stm32u5/xdisplay/stm32u5a9j-dk/display_fb.c

```diff
@@ -26,21 +26,52 @@
 #include <xdisplay.h>
 #include "display_internal.h"
 #include "mpu.h"
+#include "trustzone.h"
 
 #ifdef KERNEL_MODE
 
 // Physical frame buffers in internal SRAM memory
 __attribute__((section(".fb1")))
-ALIGN_32BYTES(uint32_t physical_frame_buffer_0[PHYSICAL_FRAME_BUFFER_SIZE]);
+ALIGN_32BYTES(uint8_t physical_frame_buffer_0[PHYSICAL_FRAME_BUFFER_SIZE]);
 
 __attribute__((section(".fb2")))
-ALIGN_32BYTES(uint32_t physical_frame_buffer_1[PHYSICAL_FRAME_BUFFER_SIZE]);
+ALIGN_32BYTES(uint8_t physical_frame_buffer_1[PHYSICAL_FRAME_BUFFER_SIZE]);
 
 // The current frame buffer selector at fixed memory address
 // It's shared between bootloaders and the firmware
 __attribute__((section(".framebuffer_select"))) uint32_t current_frame_buffer =
     0;
 
+#ifdef USE_TRUSTZONE
+void display_set_unpriv_access(bool unpriv) {
```

Comment:
> We have the framebuffer sections defined in linker scripts, we could set this directly in applet.c code. We just need to add symbols to the start/end of each framebuffer, and also we need some way to tell how many framebuffers are used.
> 
> The advantage is we'd need to do this only once instead of every display driver.

I had considered this option too. However, I ultimately chose a different path because the display driver is typically best suited to manage the framebuffers. It knows where they are located and how many are needed, and it exposes this information to the application. It’s also responsible for managing the current buffer’s protection through the MPU. Additionally, since the driver is aware of whether GFXMMU is being used, it seemed natural for this responsibility to lie within the driver itself.

> In future, this will be done in secure monitor, which ideally shouldn't care about display driver at all, and the framebuffer sections are just part of memory with some setting.

In my opinion, the secure monitor would only need to manage secure and non-secure areas, leaving privileged/unprivileged distinctions to the kernel. From the secure monitor's point of view, the SRAM used for framebuffers would be treated as unsecured memory, with its purpose defined by the kernel.